src/main/java/dev/morling/jfrunit/JfrEventTestSpockExtension.java

```diff
@@ -0,0 +1,104 @@
+/**
+ *  Copyright 2020 - 2021 The JfrUnit authors
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package dev.morling.jfrunit;
+
+import org.spockframework.runtime.extension.AbstractAnnotationDrivenExtension;
+import org.spockframework.runtime.extension.AbstractMethodInterceptor;
+import org.spockframework.runtime.extension.IMethodInvocation;
+import org.spockframework.runtime.model.FieldInfo;
+import org.spockframework.runtime.model.SpecInfo;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class JfrEventTestSpockExtension extends AbstractAnnotationDrivenExtension<JfrEventTest> {
+
+    @Override
+    public void visitSpecAnnotation(JfrEventTest annotation, SpecInfo spec) {
+        final JfrMethodInterceptor interceptor = new JfrMethodInterceptor();
+        spec.getAllFeatures().stream().forEach(featureInfo -> {
+            featureInfo.getFeatureMethod().addInterceptor(interceptor);
+        });
+    }
+
+    private String getEnabledConfiguration(IMethodInvocation invocation) {
+        return Optional.ofNullable(invocation.getMethod().getAnnotation(EnableConfiguration.class))
+                .map(EnableConfiguration::value)
+                .map(String::trim)
+                .orElse(null);
+    }
+
+    private List<EventConfiguration> getEnabledEvents(IMethodInvocation invocation) {
+        return Arrays.stream(invocation.getMethod().getReflection().getAnnotationsByType(EnableEvent.class))
+                .map(e -> new EventConfiguration(e.value(), e.stackTrace(), e.threshold(), e.period()))
+                .collect(Collectors.toList());
+    }
+
+    private List<JfrEventsContext> getJfrEvents(IMethodInvocation invocation) {
+        return invocation.getSpec().getAllFields().stream()
+                .filter(fieldInfo -> JfrEvents.class.equals(fieldInfo.getType()))
+                .map(fieldInfo -> new JfrEventsContext((JfrEvents) fieldInfo.readValue(invocation.getInstance()), fieldInfo))
+                .collect(Collectors.toList());
+    }
+
+    private class JfrMethodInterceptor extends AbstractMethodInterceptor {
```

Comment:
Why is this not a static class/in its own file?

Reply:
Yep that would be better. Changed.